### ISSUE TITLE
com.google.cloud.bigquery.BigQueryException: Unexpected error refreshing access token

### ISSUE THREAD

**Issue by manju-v** [googleapis/java-bigquery#1388]
I'm using maven package for bigquery java api:
<dependency>
			<groupId>com.google.cloud</groupId>
			<artifactId>google-cloud-bigquery</artifactId>
			<version>1.133.1</version>
</dependency>

Below is a simple java code to authenticate bigquery and query a table:
 
String projectId = BigQueryConstants.PROJECT_NAME;
    File credentialsPath = new File(BigQueryConstants.AUTH_FILENAME);
    GoogleCredentials credentials;
    try (FileInputStream serviceAccountStream = new FileInputStream(credentialsPath)) {
      credentials = ServiceAccountCredentials.fromStream(serviceAccountStream);
    }

    // Instantiate a client.
    BigQuery bigquery =
        BigQueryOptions.newBuilder()
            .setCredentials(credentials)
            .setProjectId(projectId)
            .build()
            .getService();
    System.out.println(bigquery);
    QueryJobConfiguration queryConfig =
        QueryJobConfiguration.newBuilder(
            "SELECT table_name, creation_time"
                + " FROM `"+BigQueryConstants.PROJECT_NAME+"."+BigQueryConstants.DATASET_NAME 
                +".INFORMATION_SCHEMA.TABLES`"
                + " where creation_time > TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 1 HOUR)"
                )
            .setUseLegacySql(false)
            .build();
    // Print all pages of the results.
    for (FieldValueList row : bigquery.query(queryConfig).iterateAll()) {
      // String type
      sidList.add(Integer.parseInt(StringSplitter.splitToArray(
          row.get("table_name").getStringValue(), '_')[1]));
    }

Im getting exception :
in thread "main" com.google.cloud.bigquery.BigQueryException: Unexpected error refreshing access token
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.translate(HttpBigQueryRpc.java:115)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.queryRpc(HttpBigQueryRpc.java:637)
	at com.google.cloud.bigquery.BigQueryImpl$34.call(BigQueryImpl.java:1254)
	at com.google.cloud.bigquery.BigQueryImpl$34.call(BigQueryImpl.java:1251)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
	at com.google.cloud.bigquery.BigQueryImpl.queryRpc(BigQueryImpl.java:1250)
	at com.google.cloud.bigquery.BigQueryImpl.query(BigQueryImpl.java:1238)
	at com.shieldsquare.shiedflow.job.AutoDetect.start(AutoDetectSid.java:81)
	at com.shieldsquare.shiedflow.job.AutoDetect.run(AutoDetectSid.java:30)
	at com.shieldsquare.shieldflow.App.main(App.java:91)
Caused by: java.io.IOException: Unexpected error refreshing access token
	at com.google.auth.oauth2.OAuth2Credentials.unwrapDirectFuture(OAuth2Credentials.java:309)
	at com.google.auth.oauth2.OAuth2Credentials.getRequestMetadata(OAuth2Credentials.java:156)
	at com.google.auth.oauth2.ServiceAccountCredentials.getRequestMetadata(ServiceAccountCredentials.java:962)
	at com.google.auth.http.HttpCredentialsAdapter.initialize(HttpCredentialsAdapter.java:96)
	at com.google.cloud.http.HttpTransportOptions$1.initialize(HttpTransportOptions.java:159)
	at com.google.api.client.http.HttpRequestFactory.buildRequest(HttpRequestFactory.java:91)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.buildHttpRequest(AbstractGoogleClientRequest.java:404)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:514)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:455)
	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:565)
	at com.google.cloud.bigquery.spi.v2.HttpBigQueryRpc.queryRpc(HttpBigQueryRpc.java:635)
	... 10 more
Caused by: java.lang.NoSuchMethodError: com.google.common.io.ByteStreams.exhaust(Ljava/io/InputStream;)J
	at com.google.api.client.http.ConsumingInputStream.close(ConsumingInputStream.java:40)
	at java.util.zip.InflaterInputStream.close(Unknown Source)
	at java.util.zip.GZIPInputStream.close(Unknown Source)
	at sun.nio.cs.StreamDecoder.implClose(Unknown Source)
	at sun.nio.cs.StreamDecoder.close(Unknown Source)
	at java.io.InputStreamReader.close(Unknown Source)
	at com.google.gson.stream.JsonReader.close(JsonReader.java:1218)
	at com.google.api.client.json.gson.GsonParser.close(GsonParser.java:53)
	at com.google.api.client.json.JsonParser.parse(JsonParser.java:363)
	at com.google.api.client.json.JsonParser.parse(JsonParser.java:335)
	at com.google.api.client.json.JsonObjectParser.parseAndClose(JsonObjectParser.java:79)
	at com.google.api.client.json.JsonObjectParser.parseAndClose(JsonObjectParser.java:73)
	at com.google.api.client.http.HttpResponse.parseAs(HttpResponse.java:456)
	at com.google.auth.oauth2.ServiceAccountCredentials.refreshAccessToken(ServiceAccountCredentials.java:618)
	at com.google.auth.oauth2.OAuth2Credentials$1.call(OAuth2Credentials.java:243)
	at com.google.auth.oauth2.OAuth2Credentials$1.call(OAuth2Credentials.java:240)
	at java.util.concurrent.FutureTask.run(Unknown Source)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:456)
	at com.google.auth.oauth2.OAuth2Credentials$AsyncRefreshResult.executeIfNew(OAuth2Credentials.java:567)
	at com.google.auth.oauth2.OAuth2Credentials.asyncFetch(OAuth2Credentials.java:206)
	... 20 more

**Comment by stephaniewang526** [comment]
This could mean anything like maybe something is problematic with your credentials. Please verify that your credentials are working.

**Comment by manju-v** [comment]
Hi, Thanks for your quick response. Yes credentials looks fine. I have given Bigquery.User and Bigquery.Owner, Editor and MetaDataViwer roles. I was using information_schema.tables in the select query any special permission required other than what I have mentioned here?

**Comment by stephaniewang526** [comment]
Please check with your database admin in the type of credentials you would need to access the tables.

**Comment by manju-v** [comment]
I found the issue is not with the credentials instead with the guava version I have in my pom which is older and not having a preconditions method with args. And the refresh access token exception is completely misleading as if its issue with credentials.

I tried with google-cloud-bigquery1.35.1 and it showed me the exact error and helped me debug root cause of the issue. Is it possible to correct such misleading exception and instead throw exact exception that has occured.

Thanks in advance!
